lib/src/entrypoint.dart

```diff
@@ -620,12 +620,16 @@ class Entrypoint {
       }
 
       final source = cache.source(lockFileId.source);
-      final lockFilePackagePath =
-          p.join(root.dir, source.getDirectory(lockFileId));
+      final lockFilePackagePath = root.path(source.getDirectory(lockFileId));
 
       // Make sure that the packagePath agrees with the lock file about the
       // path to the package.
-      if (p.normalize(packagePath) != p.normalize(lockFilePackagePath)) {
+      //
+      // TODO(sigurdm): We use the absolute paths to work around:
+      // https://github.com/dart-lang/package_config/issues/99
+      // For the case where Flutter has rewritten package_config.json.
+      if (p.absolute(p.normalize(packagePath)) !=
+          p.absolute(p.normalize(lockFilePackagePath))) {
```

Comment:
We could consider only allowing this if `flutter_gen` is present? Just pass a boolean `allowRelativePaths` to `_isPackagePathsMappingUpToDateWithLockfile`?

Reply:
We won't do this anyway